src/radical/pilot/utils/session.py

```diff
@@ -123,14 +123,19 @@ def fetch_profiles (sid, dburl=None, client=None, tgt=None, access=None, session
         except saga.DoesNotExist:
             print "exception(TODO): profiles tarball doesnt exists!"
 
+        try:
+            os.mkdir("%s/%s" % (tgt_url.path, pilot['_id']))
+        except OSError:
+            pass
+
         # We now have a local tarball
         if tarball_available:
             print "Extracting tarball %s into '%s'." % (ftgt.path, tgt_url.path)
             tarball = tarfile.open(ftgt.path)
-            tarball.extractall(tgt_url.path)
+            tarball.extractall("%s/%s" % (tgt_url.path, pilot['_id']))
```

Comment:
Ah right, good catch to handle multiple pilots. This likely requires changes in the consumption parts too.